src/main/scala/org/qslib/quantscale/InterestRate.scala

```diff
@@ -41,20 +41,44 @@ package org.qslib.quantscale
 import org.qslib.quantscale.time.DayCounter
 import org.qslib.quantscale.time._
 import com.github.nscala_time.time.Imports._
+import org.qslib.quantscale.time.Frequency.{Once, NoFrequency}
 
-sealed trait Compounding
+sealed trait Compounding {
+  def factor(rate: Rate, frequency: Frequency, time: Time): Real
 
+  def rate(compound: Real, frequency: Frequency, time: Time): Real
+}
 /** 1+rt */
-case object Simple extends Compounding
+case object Simple extends Compounding {
+  override def factor(rate: Rate, frequency: Frequency, time: Time): Real = 1.0 + rate * time
+
+  override def rate(compound: Real, frequency: Frequency, time: Time): Real = (compound - 1.0) / time
+}
 
 /** (1+r)^t */
-case object Compounded extends Compounding
+case object Compounded extends Compounding {
+  override def factor(rate: Rate, frequency: Frequency, time: Time): Real = Math.pow(1.0 + rate / frequency, frequency * time)
+
+  override def rate(compound: Real, frequency: Frequency, time: Time): Real = (Math.pow(compound, 1.0 / (frequency * time)) - 1.0) * frequency
+}
 
 /** e^(rt) */
-case object Continuous extends Compounding
+case object Continuous extends Compounding {
+  override def factor(rate: Rate, frequency: Frequency, time: Time): Real = Math.exp(rate * time)
+
+  override def rate(compound: Real, frequency: Frequency, time: Time): Real = Math.log(compound) / time
+}
 
 /** Simple up to the first period then Compounded */
-case object SimpleThenCompounded extends Compounding
+case object SimpleThenCompounded extends Compounding {
```

Comment:
Consider splitting this into two separate `object`s instead of repeating the if-condition.